src/ast.ml

```diff
@@ -661,39 +665,41 @@ and
   | Inductive of  (* inductief data type regel-naam-type parameters-lijst van constructors*)
       loc *
       string *
-      string list *
+      (string * ghostness) list * (*tparams*)
       ctor list
   | AbstractTypeDecl of loc * string
   | Class of
       loc *
       bool (* abstract *) *
       class_finality *
-      string *
+      string * (* class name *)
       meth list *
       field list *
       cons list *
-      string (* superclass *) *
-      string list (* itfs *) *
+      (string * (string * ghostness) list) (* superclass *) *
+      (string * ghostness) list (* type parameters *) *
+      (string * (string * ghostness) list) list (* itfs  with passed tparams*) *
       instance_pred_decl list
   | Interface of 
       loc *
       string *
-      string list *
+      (string * (string * ghostness) list) list * (* extended interfaces with passed tparams *)
       field list *
       meth list *
+      (string * ghostness) list * (* type parameters *) 
       instance_pred_decl list
   | PredFamilyDecl of
       loc *
       string *
-      string list (* type parameters *) *
+      (string * ghostness) list (* type parameters *) *

```

Comment:
In the interest of minimality, it would have been better to not add support for real type parameters on predicates for now?